database/bootstrap.go

```diff
@@ -91,13 +92,18 @@ func BootstrapDqlite(
 	if err != nil {
 		return errors.Annotatef(err, "opening controller database")
 	}
+
+	if err := pragma.SetPragma(ctx, db, pragma.ForeignKeysPragma, true); err != nil {
+		return errors.Annotate(err, "setting foreign keys pragma")
```

Comment:
My preference here and I know this style is more preferred by some but to use `fmt.Errorf`. Using annotate still brings cause into play and creates divergent errors paths that can be complicated to use with Is and As.

Not to be militant about it.